src/Slacklinker/Linear/Types.hs

```diff
@@ -0,0 +1,49 @@
+-- | Types for the Slacklinker Linear integration
+module Slacklinker.Linear.Types (
+  LinearClientId (..),
+  LinearClientSecret (..),
+  LinearCreds (..),
+  LinearBearerToken (..),
+  LinearTicketUUID (..),
+  LinearTicketId (..),
+  linearTicketIdToText,
+) where
+
+import Database.Persist.Class (PersistField)
+import Database.Persist.Sql (PersistFieldSql)
+import Slacklinker.Prelude
+import Slacklinker.UUID (UUID)
+
+newtype LinearClientId = LinearClientId {unLinearClientId :: Text}
+  deriving stock (Show)
+
+newtype LinearClientSecret = LinearClientSecret {unLinearClientSecret :: Text}
+
+instance Show LinearClientSecret where
+  show _ = "LinearClientSecret REDACTED"
```

Comment:
Non-blocking: Maybe we should make a [`secrecy`](https://docs.rs/secrecy/latest/secrecy/) equivalent for Haskell?